spark3/src/main/java/org/apache/iceberg/spark/source/StreamingOffset.java

```diff
@@ -62,20 +63,23 @@ static StreamingOffset fromJson(String json) {
 
     try {
       JsonNode node = JsonUtil.mapper().readValue(json, JsonNode.class);
-      // The version of StreamingOffset. The offset was created with a version number
-      // used to validate when deserializing from json string.
-      int version = JsonUtil.getInt(VERSION, node);
-      Preconditions.checkArgument(version == CURR_VERSION,
-              "Cannot parse offset JSON: offset version %s is not supported", version);
+      return fromJsonNode(node);
+    } catch (IOException e) {
+      throw new IllegalArgumentException(String.format("Failed to parse StreamingOffset from JSON string %s", json), e);
```

Comment:
I think this should also be `UncheckedIOException`.

Reply:
fixed it.